### PR TITLE
OPENSSL_cpuid_setup FreeBSD PowerPC update

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

File: crypto/ppccap.c [hunk 1]
```diff
@@ -212,6 +212,22 @@ size_t OPENSSL_instrument_bus2(unsigned int *out, size_t cnt, size_t max)
 #  include <sys/auxv.h>
 #  define OSSL_IMPLEMENT_GETAUXVAL
 # endif
+# if defined(__FreeBSD__)
```
Comment (mattcaswell) [inline crypto/ppccap.c:215]: Should this be an "else" from the previous "if"?

Comment (devnexen) [reply crypto/ppccap.c:215]: no in fact wrong level on my part.

File: crypto/ppccap.c [hunk 2]
```diff
@@ -214,6 +214,24 @@ size_t OPENSSL_instrument_bus2(unsigned int *out, size_t cnt, size_t max)
 # endif
 #endif
 
+#if defined(__FreeBSD__)
```
Comment (mattcaswell) [inline crypto/ppccap.c:217]: Is there a possibility that `OSSL_IMPLEMENT_GETAUXVAL` could already be defined via the block above? And if so, should we have a check for that below so we don't attempt to define it twice?

Comment (devnexen) [reply crypto/ppccap.c:217]: Nope the check above is about Linux/GLIBC cases.

**Review by mattcaswell — APPROVED** [review]

**Review by kaduk — APPROVED** [review]
@mattcaswell perhaps you could look at #13650 which is the same thing but for armcap?

**Comment by mattcaswell** [comment]
Pushed. Thanks.

**Pull request closed without merging**